src/tigerbeetle-java/src/main/java/com/tigerbeetle/Account.java

```diff
@@ -50,99 +56,208 @@ public Account() {
         timestamp = ptr.getLong();
     }
 
-    public short getCode() {
+    /**
+     * Gets an identifier for this account, defined by the user.
+     *
+     * @return An {@link java.util.UUID} representing an integer-encoded UUIDv4 or any other unique
+     *         128-bit integer.
+     */
+    public UUID getId() {
+        return id;
+    }
+
+    /**
+     * Sets an identifier for this account, defined by the user.
+     * <p>
+     * Must be unique and non-zero.
+     *
+     * @param id An {@link java.util.UUID} representing an integer-encoded UUIDv4 or any other
+     *        unique 128-bit integer.
+     * @throws NullPointerException if {@code id} is null.
+     */
+    public void setId(UUID id) {
```

Comment:
It is not required that the id is a UUID. So I'm not sure we should do this.

Same for get/setUserData. UUID is *one* valid kind of id and userData but it is not the only one. It is just an example.

Reply:
100% — it should be a 128-bit integer only as far as possible. The most primitive type.